api/v1beta1/nutanixcluster_types.go

```diff
@@ -46,14 +46,12 @@ type NutanixClusterSpec struct {
 
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	// host can be either DNS name or ip address
-	// +optional
 	ControlPlaneEndpoint capiv1.APIEndpoint `json:"controlPlaneEndpoint"`
 
 	// prismCentral holds the endpoint address and port to access the Nutanix Prism Central.
 	// When a cluster-wide proxy is installed, by default, this endpoint will be accessed via the proxy.
 	// Should you wish for communication with this endpoint not to be proxied, please add the endpoint to the
 	// proxy spec.noProxy list.
-	// +optional
 	PrismCentral *credentialTypes.NutanixPrismEndpoint `json:"prismCentral"`
```

Comment:
This should not be a pointer if its required

Reply:
could you please elaborate on why it should not be a pointer to make it required.
if we change this then we will have to upgrade api version from v1beta1 to v1beta2 else we will lose backward compatibility.

Reply:
makes sense to not be a pointer type per convention documented in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required 
<pre>
Note that for backward compatibility, any field that has the omitempty struct tag will be considered to be optional, but this may change in the future and having the +optional comment tag is highly recommended.

Required fields have the opposite properties, namely:

They do not have an +optional comment tag.
They do not have an omitempty struct tag.
They are not a pointer type in the Go definition (e.g. AnotherFlag SomeFlag).
The API server should not allow POSTing or PUTing a resource with this field unset.
</pre>

but that will break the backward compatibility hence keeping it as is. on contract upgrade, we can do the same